app/src/main/java/de/westnordost/streetcomplete/data/QuestController.java

```diff
@@ -178,16 +202,26 @@ public boolean createNote(long osmQuestId, String questTitle, String text, @Null
 
 		osmElementDB.deleteUnreferenced();
 		geometryDB.deleteUnreferenced();
-		return true;
 	}
 
-	public void createNote(String text, @Nullable List<String> imagePaths, LatLon position)
+	/** Split a way for the given OSM Quest. The quest will turn invisible.
+	 *  @return true if successful */
+	public boolean splitWay(long osmQuestId, @NonNull List<? extends SplitPolylineAtPosition> splits, @NonNull String source)
 	{
-		CreateNote createNote = new CreateNote();
-		createNote.position = position;
-		createNote.text = text;
-		createNote.imagePaths = imagePaths;
-		createNoteDB.add(createNote);
+		OsmQuest q = osmQuestDB.get(osmQuestId);
+		// race condition: another thread may have removed the element already (#288)
+		if(q == null || q.getStatus() != QuestStatus.NEW) return false;
+
```

Comment:
It seems to me that race condition may happen anyway - with another thread acting after this check and before next instructions.

Reply:
Theoretically yes, but what thread should that be? `splitWay` is called by user interaction, the user is not able to quickly answer the quest while this method is executed, because this method is executed on the main thread as well. It doesn't look like the fact that it is executed on the main thread creates a noticeable hiccup in the UI but if you are interested, you could measure how long it takes for this method to finish. 

Though, even if it wouldn't be on the main thread, the user would not have enough time to quickly answer the question after all while this method executes. If the upload thread removes the quest in the meantime because the element changed substantially, it would also be no problem, because the `OsmQuestSplitWay` is created and saved to be uploaded later, but on upload, it will simply fail with a conflict then.